web/src/app/lists/FlatList.tsx

```diff
@@ -0,0 +1,289 @@
+import React from 'react'
+import List from '@material-ui/core/List'
+import ListItem from '@material-ui/core/ListItem'
+import ListItemIcon from '@material-ui/core/ListItemIcon'
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
+import ListItemText from '@material-ui/core/ListItemText'
+import Typography from '@material-ui/core/Typography'
+import {
+  DragDropContext,
+  Droppable,
+  Draggable,
+  DropResult,
+  DraggableStateSnapshot,
+  DraggableProvided,
+  DroppableProvided,
+} from 'react-beautiful-dnd'
+import ListSubheader from '@material-ui/core/ListSubheader'
+import { AppLink } from '../util/AppLink'
+import { makeStyles } from '@material-ui/core'
+import { CSSTransition, TransitionGroup } from 'react-transition-group'
+
+const lime = '#93ed94'
+const lightLime = '#defadf'
+const lightGrey = '#ebebeb'
+
+const useStyles = makeStyles({
+  background: { backgroundColor: 'white' },
+  highlightedItem: {
+    width: '100%',
+    borderLeft: '6px solid ' + lime,
+    background: lightLime,
+  },
+  participantDragging: {
+    backgroundColor: lightGrey,
+  },
+  slideEnter: {
+    transform: 'translateX(-100%)',
+  },
+  slideEnterActive: {
+    transform: 'translateX(0%)',
+    transition: 'opacity 500ms, transform 500ms',
+  },
+  slideExit: {
+    transform: 'translateX(0%)',
+  },
+  slideExitActive: {
+    transform: 'translateX(-100%)',
+    transition: 'opacity 500ms, transform 500ms',
+  },
+  listItem: {
+    width: '100%',
+  },
+  listItemText: {
+    fontStyle: 'italic',
+  },
+})
+
+export interface FlatListSub {
+  subHeader: string
+}
+export interface FlatListItem {
+  title: string
+  highlight?: boolean
+  subText?: JSX.Element | string
+  icon?: JSX.Element
+  secondaryAction?: JSX.Element | null
+  url?: string
+  id?: string
+}
+
+export type FlatListListItem = FlatListSub | FlatListItem
+
+export interface FlatListType {
+  items: FlatListListItem[]
+
+  // headerNote will be displayed at the top of the list.
+  headerNote?: string
+
+  // emptyMessage will be displayed if there are no items in the list.
+  emptyMessage?: string
+
+  listProps?: { dense: boolean }
+
+  // indent text of each list item if no icon is present
+  inset?: boolean
+
+  // If specified, enables drag and drop
+  //
+  // onReorder(id, oldIndex, newIndex)
+  onReorder?: (oldIndex: number, newIndex: number) => void
+
+  // will render transition in list
+  transition?: boolean
+}
+
+export default function FlatList(props: FlatListType): JSX.Element {
```

Comment:
nit: we've been naming interfaces/types with "Props" appended to the end
```suggestion
export default function FlatList(props: FlatListProps): JSX.Element {
```